{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "\n",
    "import scipy.stats as stats\n",
    "\n",
    "import pickle as pl\n",
    "\n",
    "from collections import Counter\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "#from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get co-factor, binding site, etc from uniprot annotation file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Gene names  (ordered locus )</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Length</th>\n",
       "      <th>Absorption</th>\n",
       "      <th>Active site</th>\n",
       "      <th>Activity regulation</th>\n",
       "      <th>Binding site</th>\n",
       "      <th>Calcium binding</th>\n",
       "      <th>Catalytic activity</th>\n",
       "      <th>Cofactor</th>\n",
       "      <th>DNA binding</th>\n",
       "      <th>Metal binding</th>\n",
       "      <th>Nucleotide binding</th>\n",
       "      <th>Site</th>\n",
       "      <th>Motif</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Entry</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P00729</th>\n",
       "      <td>CBPY_YEAST</td>\n",
       "      <td>YMR297W</td>\n",
       "      <td>PRC1 YMR297W</td>\n",
       "      <td>532</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACT_SITE 257 257 {ECO:0000269|PubMed:7727362, ...</td>\n",
       "      <td>ACTIVITY REGULATION: Inhibited by ZPCK.</td>\n",
       "      <td>BINDING 452 452 Substrate. {ECO:0000305|Ref.5}...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CATALYTIC ACTIVITY: Reaction=Release of a C-te...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MOTIF 24 27 Vacuolar targeting signal. {ECO:00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q08054</th>\n",
       "      <td>CSI2_YEAST</td>\n",
       "      <td>YOL007C</td>\n",
       "      <td>CSI2 YOL007C</td>\n",
       "      <td>341</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P38845</th>\n",
       "      <td>CRP1_YEAST</td>\n",
       "      <td>YHR146W</td>\n",
       "      <td>CRP1 YHR146W</td>\n",
       "      <td>465</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P39988</th>\n",
       "      <td>BUD16_YEAST</td>\n",
       "      <td>YEL029C</td>\n",
       "      <td>BUD16 YEL029C</td>\n",
       "      <td>312</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BINDING 9 9 Substrate. {ECO:0000250}.; BINDING...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CATALYTIC ACTIVITY: Reaction=ATP + pyridoxal =...</td>\n",
       "      <td>COFACTOR: Name=a divalent metal cation; Xref=C...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_BIND 183 184 ATP. {ECO:0000250}.; NP_BIND 2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q12289</th>\n",
       "      <td>CRC1_YEAST</td>\n",
       "      <td>YOR100C</td>\n",
       "      <td>CRC1 YOR100C YOR3193C</td>\n",
       "      <td>327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P32873</th>\n",
       "      <td>BEM3_YEAST</td>\n",
       "      <td>YPL115C</td>\n",
       "      <td>BEM3 YPL115C LPH12C</td>\n",
       "      <td>1128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P53730</th>\n",
       "      <td>ALG12_YEAST</td>\n",
       "      <td>YNR030W</td>\n",
       "      <td>ALG12 ECM39 YNR030W N3265</td>\n",
       "      <td>551</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CATALYTIC ACTIVITY: Reaction=alpha-D-Man-(1-&gt;2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q12291</th>\n",
       "      <td>BMT6_YEAST</td>\n",
       "      <td>YLR063W</td>\n",
       "      <td>BMT6 YLR063W L2174</td>\n",
       "      <td>365</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CATALYTIC ACTIVITY: Reaction=S-adenosyl-L-meth...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q05123</th>\n",
       "      <td>ARP9_YEAST</td>\n",
       "      <td>YMR033W</td>\n",
       "      <td>ARP9 SWP59 YMR033W YM9973.07</td>\n",
       "      <td>467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P32623</th>\n",
       "      <td>CRH2_YEAST</td>\n",
       "      <td>YEL040W</td>\n",
       "      <td>UTR2 CRH2 YEL040W SYGP-ORF18</td>\n",
       "      <td>467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACT_SITE 166 166 Nucleophile. {ECO:0000250}.; ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Entry name Gene names  (ordered locus )  \\\n",
       "Entry                                              \n",
       "P00729   CBPY_YEAST                      YMR297W   \n",
       "Q08054   CSI2_YEAST                      YOL007C   \n",
       "P38845   CRP1_YEAST                      YHR146W   \n",
       "P39988  BUD16_YEAST                      YEL029C   \n",
       "Q12289   CRC1_YEAST                      YOR100C   \n",
       "P32873   BEM3_YEAST                      YPL115C   \n",
       "P53730  ALG12_YEAST                      YNR030W   \n",
       "Q12291   BMT6_YEAST                      YLR063W   \n",
       "Q05123   ARP9_YEAST                      YMR033W   \n",
       "P32623   CRH2_YEAST                      YEL040W   \n",
       "\n",
       "                          Gene names  Length Absorption  \\\n",
       "Entry                                                     \n",
       "P00729                  PRC1 YMR297W     532        NaN   \n",
       "Q08054                  CSI2 YOL007C     341        NaN   \n",
       "P38845                  CRP1 YHR146W     465        NaN   \n",
       "P39988                 BUD16 YEL029C     312        NaN   \n",
       "Q12289         CRC1 YOR100C YOR3193C     327        NaN   \n",
       "P32873           BEM3 YPL115C LPH12C    1128        NaN   \n",
       "P53730     ALG12 ECM39 YNR030W N3265     551        NaN   \n",
       "Q12291            BMT6 YLR063W L2174     365        NaN   \n",
       "Q05123  ARP9 SWP59 YMR033W YM9973.07     467        NaN   \n",
       "P32623  UTR2 CRH2 YEL040W SYGP-ORF18     467        NaN   \n",
       "\n",
       "                                              Active site  \\\n",
       "Entry                                                       \n",
       "P00729  ACT_SITE 257 257 {ECO:0000269|PubMed:7727362, ...   \n",
       "Q08054                                                NaN   \n",
       "P38845                                                NaN   \n",
       "P39988                                                NaN   \n",
       "Q12289                                                NaN   \n",
       "P32873                                                NaN   \n",
       "P53730                                                NaN   \n",
       "Q12291                                                NaN   \n",
       "Q05123                                                NaN   \n",
       "P32623  ACT_SITE 166 166 Nucleophile. {ECO:0000250}.; ...   \n",
       "\n",
       "                            Activity regulation  \\\n",
       "Entry                                             \n",
       "P00729  ACTIVITY REGULATION: Inhibited by ZPCK.   \n",
       "Q08054                                      NaN   \n",
       "P38845                                      NaN   \n",
       "P39988                                      NaN   \n",
       "Q12289                                      NaN   \n",
       "P32873                                      NaN   \n",
       "P53730                                      NaN   \n",
       "Q12291                                      NaN   \n",
       "Q05123                                      NaN   \n",
       "P32623                                      NaN   \n",
       "\n",
       "                                             Binding site Calcium binding  \\\n",
       "Entry                                                                       \n",
       "P00729  BINDING 452 452 Substrate. {ECO:0000305|Ref.5}...             NaN   \n",
       "Q08054                                                NaN             NaN   \n",
       "P38845                                                NaN             NaN   \n",
       "P39988  BINDING 9 9 Substrate. {ECO:0000250}.; BINDING...             NaN   \n",
       "Q12289                                                NaN             NaN   \n",
       "P32873                                                NaN             NaN   \n",
       "P53730                                                NaN             NaN   \n",
       "Q12291                                                NaN             NaN   \n",
       "Q05123                                                NaN             NaN   \n",
       "P32623                                                NaN             NaN   \n",
       "\n",
       "                                       Catalytic activity  \\\n",
       "Entry                                                       \n",
       "P00729  CATALYTIC ACTIVITY: Reaction=Release of a C-te...   \n",
       "Q08054                                                NaN   \n",
       "P38845                                                NaN   \n",
       "P39988  CATALYTIC ACTIVITY: Reaction=ATP + pyridoxal =...   \n",
       "Q12289                                                NaN   \n",
       "P32873                                                NaN   \n",
       "P53730  CATALYTIC ACTIVITY: Reaction=alpha-D-Man-(1->2...   \n",
       "Q12291  CATALYTIC ACTIVITY: Reaction=S-adenosyl-L-meth...   \n",
       "Q05123                                                NaN   \n",
       "P32623                                                NaN   \n",
       "\n",
       "                                                 Cofactor DNA binding  \\\n",
       "Entry                                                                   \n",
       "P00729                                                NaN         NaN   \n",
       "Q08054                                                NaN         NaN   \n",
       "P38845                                                NaN         NaN   \n",
       "P39988  COFACTOR: Name=a divalent metal cation; Xref=C...         NaN   \n",
       "Q12289                                                NaN         NaN   \n",
       "P32873                                                NaN         NaN   \n",
       "P53730                                                NaN         NaN   \n",
       "Q12291                                                NaN         NaN   \n",
       "Q05123                                                NaN         NaN   \n",
       "P32623                                                NaN         NaN   \n",
       "\n",
       "       Metal binding                                 Nucleotide binding Site  \\\n",
       "Entry                                                                          \n",
       "P00729           NaN                                                NaN  NaN   \n",
       "Q08054           NaN                                                NaN  NaN   \n",
       "P38845           NaN                                                NaN  NaN   \n",
       "P39988           NaN  NP_BIND 183 184 ATP. {ECO:0000250}.; NP_BIND 2...  NaN   \n",
       "Q12289           NaN                                                NaN  NaN   \n",
       "P32873           NaN                                                NaN  NaN   \n",
       "P53730           NaN                                                NaN  NaN   \n",
       "Q12291           NaN                                                NaN  NaN   \n",
       "Q05123           NaN                                                NaN  NaN   \n",
       "P32623           NaN                                                NaN  NaN   \n",
       "\n",
       "                                                    Motif  \n",
       "Entry                                                      \n",
       "P00729  MOTIF 24 27 Vacuolar targeting signal. {ECO:00...  \n",
       "Q08054                                                NaN  \n",
       "P38845                                                NaN  \n",
       "P39988                                                NaN  \n",
       "Q12289                                                NaN  \n",
       "P32873                                                NaN  \n",
       "P53730                                                NaN  \n",
       "Q12291                                                NaN  \n",
       "Q05123                                                NaN  \n",
       "P32623                                                NaN  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_source = './uniprot-proteome_UP000002311.tab'\n",
    "\n",
    "uniprot_df = pd.read_csv(uniprot_source, sep='\\t', index_col=0)\n",
    "\n",
    "uniprot_df.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5590, 6049, 459, 'number of metal binding annots')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get an idea of the number of annotations in each category\n",
    "\n",
    "n_muts_unknown = uniprot_df['Metal binding'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of metal binding annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5590, 6049, 459, 'number of metal binding annots')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['Metal binding'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of metal binding annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5372, 6049, 677, 'number of nucleotide binding annots')"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['Nucleotide binding'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of nucleotide binding annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5795, 6049, 254, 'number of site annots')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['Site'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of site annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5923, 6049, 126, 'number of DNA binding annots')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['DNA binding'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of DNA binding annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6038, 6049, 11, 'number of Calcium binding annots')"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['Calcium binding'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of Calcium binding annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5406, 6049, 643, 'number of binding site annots')"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['Binding site'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of binding site annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5250, 6049, 799, 'number of active site annots')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['Active site'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of active site annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5628, 6049, 421, 'number of Motifs annots')"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_muts_unknown = uniprot_df['Motif'].isna().sum()\n",
    "n_entries = len(uniprot_df.index)\n",
    "\n",
    "n_muts_known = n_entries-n_muts_unknown\n",
    "\n",
    "n_muts_unknown, n_entries, n_muts_known, 'number of Motifs annots'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"MOTIF 258 268 'HIGH' region.; MOTIF 495 499 'KMSKS' region.\""
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_df.loc['P13188']['Motif']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Entry\n",
      "P39988    NP_BIND 183 184 ATP. {ECO:0000250}.; NP_BIND 2...\n",
      "P40416    NP_BIND 469 480 ATP. {ECO:0000255|PROSITE-ProR...\n",
      "Q03957    NP_BIND 189 197 ATP. {ECO:0000255|PROSITE-ProR...\n",
      "P47177    NP_BIND 41 43 FMN. {ECO:0000250}.; NP_BIND 270...\n",
      "P06243    NP_BIND 39 47 ATP. {ECO:0000255|PROSITE-ProRul...\n",
      "P53727    NP_BIND 190 191 ATP. {ECO:0000250}.; NP_BIND 2...\n",
      "P40471            NP_BIND 13 37 NAD or NADP. {ECO:0000250}.\n",
      "P32317                  NP_BIND 126 133 ATP. {ECO:0000255}.\n",
      "P53630                    NP_BIND 18 26 ATP. {ECO:0000250}.\n",
      "P39925                  NP_BIND 328 335 ATP. {ECO:0000255}.\n",
      "Name: Nucleotide binding, dtype: object\n"
     ]
    }
   ],
   "source": [
    "# the goal will be to start with these annotations and build a dataframe that has the following\n",
    "# format:\n",
    "# sys_name, aa_involved_annot_type_1, aa_involved_annot_type_2, ...\n",
    "\n",
    "# overview of what the metal binding looks like\n",
    "select_no_info = uniprot_df['Nucleotide binding'].isna() == True\n",
    "\n",
    "with_metal_bind = uniprot_df[~select_no_info]\n",
    "\n",
    "print(with_metal_bind['Nucleotide binding'].head(10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "select_no_info = uniprot_df['Metal binding'].isna() == True\n",
    "\n",
    "with_metal_bind = list(uniprot_df[~select_no_info].index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Entry name', 'Gene names  (ordered locus )', 'Gene names', 'Length',\n",
       "       'Absorption', 'Active site', 'Activity regulation', 'Binding site',\n",
       "       'Calcium binding', 'Catalytic activity', 'Cofactor', 'DNA binding',\n",
       "       'Metal binding', 'Nucleotide binding', 'Site', 'Motif'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import experimental data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_data = pd.read_csv('./screen_data_analysis/screen_guides_wscoreannot_revisedv2.tsv', \n",
    "                       sep='\\t', index_col = 0, low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_uniprot_annot(uniprot_ID, annot_type):\n",
    "    \"\"\"Goes through a uniprot annotation and returns a list of amino acids annotated to it if\n",
    "    possible.\n",
    "    uniprot_ID: the uniprot ID\n",
    "    annot_type: the Uniprot header\n",
    "    \"\"\"\n",
    "    entry = uniprot_df.loc[uniprot_ID][annot_type]\n",
    "    \n",
    "    pos_list = []\n",
    "        \n",
    "    if entry not in [np.NaN]:\n",
    "        \n",
    "        #print(entry)\n",
    "\n",
    "        if '; ' in entry and entry.count('}')>1:\n",
    "            \n",
    "            if '.; ' in entry:\n",
    "                entry = entry.split('.; ')\n",
    "            \n",
    "            else:\n",
    "                entry = entry.split('; ')      \n",
    "                \n",
    "        else:\n",
    "            entry = [entry]\n",
    "\n",
    "        #print(entry)\n",
    "        for annot in entry:\n",
    "            #print(repr(annot))\n",
    "            annot_components = annot.split(' ')\n",
    "            #print(annot_components)\n",
    "            annot_start = int(annot_components[1].strip(';'))\n",
    "            annot_end = int(annot_components[2].strip(';'))\n",
    "\n",
    "            if annot_start != annot_end:\n",
    "                #print(annot_components)\n",
    "                #print(annot_start, annot_end)\n",
    "\n",
    "                aa_in_annot = range(annot_start, annot_end+1)\n",
    "                # the +1 is added to make the range inclusive\n",
    "                aa_to_add = [x for x in aa_in_annot if x not in pos_list]\n",
    "\n",
    "                pos_list += aa_to_add\n",
    "\n",
    "            if annot_start == annot_end:\n",
    "                if annot_start not in pos_list:\n",
    "                    pos_list.append(annot_start)\n",
    "        #print(repr(pos_list))\n",
    "\n",
    "\n",
    "            #if annot_components[0] != 'METAL':\n",
    "            #    print(annot_components[0])\n",
    "\n",
    "    return pos_list\n",
    "\n",
    "\n",
    "dict_of_annot = {}\n",
    "\n",
    "\n",
    "for uniprot_ID in list(uniprot_df.index):\n",
    "    \n",
    "    sys_name_dict = {}\n",
    "    out_dict = {}\n",
    "    \n",
    "    sys_name = uniprot_df.loc[uniprot_ID]['Gene names  (ordered locus )']\n",
    "    #print(uniprot_ID, sys_name)\n",
    "    \n",
    "    out_dict['metal'] = parse_uniprot_annot(uniprot_ID, 'Metal binding')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'Metal binding'), 'Metal')\n",
    "    \n",
    "    out_dict['Nucl'] = parse_uniprot_annot(uniprot_ID, 'Nucleotide binding')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'Nucleotide binding'), 'Nucl')\n",
    "    \n",
    "    out_dict['Site'] = parse_uniprot_annot(uniprot_ID, 'Site')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'Site'), 'Site')\n",
    "    \n",
    "    out_dict['DNA_bind'] = parse_uniprot_annot(uniprot_ID, 'DNA binding')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'DNA binding'), 'DNA bind')\n",
    "    \n",
    "    out_dict['Ca_bind'] = parse_uniprot_annot(uniprot_ID, 'Calcium binding')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'Calcium binding'), 'Ca bind')\n",
    "    \n",
    "    out_dict['bind_site'] = parse_uniprot_annot(uniprot_ID, 'Binding site')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'Binding site'), 'Bind site')\n",
    "    \n",
    "    out_dict['act_site'] = parse_uniprot_annot(uniprot_ID, 'Active site')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'Active site'), 'Active site')\n",
    "    \n",
    "    out_dict['motif'] = parse_uniprot_annot(uniprot_ID, 'Motif')\n",
    "    #print(parse_uniprot_annot(uniprot_ID, 'Motif'), 'Motif')\n",
    "    \n",
    "    #print('############')\n",
    "    #print(out_dict)\n",
    "    \n",
    "    sensitive_pos_list = list(set([int(item) for sublist in out_dict.values() for item in sublist]))\n",
    "    \n",
    "    sys_name_dict['annot_pos_list'] = sensitive_pos_list\n",
    "    \n",
    "    sys_name_dict['annots'] = out_dict\n",
    "    \n",
    "    dict_of_annot[sys_name] = sys_name_dict\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_aa_affected(mut):\n",
    "    \n",
    "    if mut == 'invalid seq' or mut == 'error':\n",
    "        \n",
    "        return 'NA'\n",
    "    \n",
    "    elif 'silent' in mut:\n",
    "        mut = mut.split('|')\n",
    "        pos = mut[3]\n",
    "        return int(pos)\n",
    "    \n",
    "    else:\n",
    "        mut = mut.split('|')\n",
    "        pos = mut[1]\n",
    "        return int(pos)\n",
    "    \n",
    "get_aa_affected('S|2|*')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_data['mut1_perturb_annot'] = np.nan\n",
    "exp_data['mut2_perturb_annot'] = np.nan\n",
    "exp_data['mut1_near_annot'] = np.nan\n",
    "exp_data['mut2_near_annot'] = np.nan\n",
    "\n",
    "exp_data['perturb_annot_info'] = 'NA'\n",
    "\n",
    "valid_ids = [x for x in dict_of_annot.keys()]\n",
    "\n",
    "\n",
    "for guide in exp_data.index:\n",
    "    \n",
    "    sys_name = exp_data.loc[guide]['target']\n",
    "    \n",
    "    pos1 = None\n",
    "    pos2 = None\n",
    "\n",
    "    mut1 = exp_data.loc[guide]['mut1_C>G']\n",
    "    pos1 = get_aa_affected(mut1)\n",
    "    \n",
    "    \n",
    "    mut2 = exp_data.loc[guide]['mut2_C>G']\n",
    "    \n",
    "    if mut2 in [np.NaN]:\n",
    "        pos2 = 'NA'\n",
    "        \n",
    "    else:\n",
    "        pos2 = get_aa_affected(mut2)\n",
    "    \n",
    "    \n",
    "    if sys_name in valid_ids:\n",
    "        \n",
    "        target_annot_list = dict_of_annot[sys_name]['annot_pos_list']\n",
    "        \n",
    "        #print (target_annot_list)\n",
    "        \n",
    "        if target_annot_list == []:\n",
    "            exp_data.at[guide, ('perturb_annot_info')] = 'Empty'\n",
    "            \n",
    "        else:\n",
    "            exp_data.at[guide, ('perturb_annot_info')] = dict_of_annot[sys_name]['annots']\n",
    "            \n",
    "            near_list = []\n",
    "            \n",
    "            for site in list(target_annot_list):\n",
    "                for offset in [-2,-1,1,2]:\n",
    "                    near_list.append((site+offset))\n",
    "                    \n",
    "        \n",
    "            if pos1 in target_annot_list:\n",
    "                exp_data.at[guide, ('mut1_perturb_annot')] = 1\n",
    "                exp_data.at[guide, ('mut1_near_annot')] = 1\n",
    "                \n",
    "            else:\n",
    "                exp_data.at[guide, ('mut1_perturb_annot')] = 0\n",
    "                \n",
    "                if pos1 in near_list:\n",
    "                    exp_data.at[guide, ('mut1_near_annot')] = 1\n",
    "                    \n",
    "                else:\n",
    "                    exp_data.at[guide, ('mut1_near_annot')] = 0\n",
    "                \n",
    "                \n",
    "                \n",
    "            if pos2 in target_annot_list:\n",
    "                exp_data.at[guide, ('mut2_perturb_annot')] = 1\n",
    "                exp_data.at[guide, ('mut2_near_annot')] = 1\n",
    "                \n",
    "            else:\n",
    "                exp_data.at[guide, ('mut2_perturb_annot')] = 0\n",
    "                exp_data.at[guide, ('mut2_near_annot')] = 0\n",
    "                \n",
    "                if pos2 in near_list:\n",
    "                    exp_data.at[guide, ('mut2_near_annot')] = 1\n",
    "                    \n",
    "                else:\n",
    "                    exp_data.at[guide, ('mut2_near_annot')] = 0\n",
    "            \n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "    #print(sys_name, pos1, pos2)\n",
    "    #print(target_annot_list)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15876 in uniprot\n",
      "695 in uniprot\n",
      "6295 in uniprot with annots\n",
      "343 in uniprot with annots, also over_threshold\n",
      "188 349\n",
      "37 54\n"
     ]
    }
   ],
   "source": [
    "n_sensitive_site = (exp_data['mut1_perturb_annot'] == 1) | (exp_data['mut2_perturb_annot'] == 1)\n",
    "n_near_sensitive_site = (exp_data['mut1_near_annot'] == 1) | (exp_data['mut2_near_annot'] == 1)\n",
    "\n",
    "in_uniprot = exp_data['perturb_annot_info'] != 'NA' \n",
    "annot_not_empty = exp_data['perturb_annot_info'] != 'Empty' \n",
    "over_threshold = exp_data['z_score_meanSE'] <=-2.01\n",
    "\n",
    "is_essential = exp_data['orf_type'] == 'essential'\n",
    "\n",
    "\n",
    "print (len(exp_data[in_uniprot]), 'in uniprot')\n",
    "print (len(exp_data[in_uniprot&over_threshold]), 'in uniprot')\n",
    "print (len(exp_data[in_uniprot&annot_not_empty]), 'in uniprot with annots')\n",
    "print (len(exp_data[in_uniprot&annot_not_empty&over_threshold]), 'in uniprot with annots, also over_threshold')\n",
    "print (len(exp_data[n_sensitive_site]), len(exp_data[n_near_sensitive_site]))\n",
    "print (len(exp_data[n_sensitive_site&over_threshold]), len(exp_data[n_near_sensitive_site&over_threshold]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 144\n",
      "17 144\n"
     ]
    }
   ],
   "source": [
    "print (len(exp_data[(~n_sensitive_site)&n_near_sensitive_site&over_threshold]), \n",
    "       len(exp_data[(~n_sensitive_site)&n_near_sensitive_site&(~over_threshold)]))\n",
    "\n",
    "print (len(exp_data[(~n_sensitive_site)&n_near_sensitive_site&over_threshold]), \n",
    "       len(exp_data[(~n_sensitive_site)&n_near_sensitive_site&(~over_threshold)]))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(89.40666738789066,\n",
       " 3.851135985457805e-20,\n",
       " 2,\n",
       " array([[  10.24368546,    7.84622716,  324.91008737],\n",
       "        [ 177.75631454,  136.15377284, 5638.08991263]]))"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obs = np.array([[37, 17, 289], [151, 127, 5674]])\n",
    "stats.chi2_contingency(obs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4.810788514860561, 1.4457178912564665e-12)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obs = np.array([[37,289], [151,5674]])\n",
    "stats.fisher_exact(obs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2.6280685502547474, 0.0007962726372044745)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obs = np.array([[17,289], [127,5674]])\n",
    "stats.fisher_exact(obs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13531 in uniprot\n",
      "1001 in uniprot\n",
      "5455 in uniprot with annots\n",
      "485 in uniprot with annots, also over_threshold\n",
      "158 140\n",
      "45 20\n"
     ]
    }
   ],
   "source": [
    "n_sensitive_site = (exp_data['mut1_perturb_annot'] == 1) | (exp_data['mut2_perturb_annot'] == 1)\n",
    "n_near_sensitive_site = (exp_data['mut1_near_annot'] == 1) | (exp_data['mut2_near_annot'] == 1)\n",
    "\n",
    "in_uniprot = exp_data['perturb_annot_info'] != 'NA' \n",
    "annot_not_empty = exp_data['perturb_annot_info'] != 'Empty' \n",
    "over_threshold = exp_data['over_threshold'] == 'Yes'\n",
    "\n",
    "is_essential = exp_data['orf_type'] == 'essential'\n",
    "\n",
    "\n",
    "print (len(exp_data[in_uniprot&is_essential]), 'in uniprot')\n",
    "print (len(exp_data[in_uniprot&over_threshold&is_essential]), 'in uniprot')\n",
    "print (len(exp_data[in_uniprot&annot_not_empty&is_essential]), 'in uniprot with annots')\n",
    "print (len(exp_data[in_uniprot&annot_not_empty&over_threshold&is_essential]), 'in uniprot with annots, also over_threshold')\n",
    "print (len(exp_data[n_sensitive_site&is_essential]), len(exp_data[(~n_sensitive_site)&n_near_sensitive_site&is_essential]))\n",
    "print (len(exp_data[n_sensitive_site&over_threshold&is_essential]), len(exp_data[(~n_sensitive_site)&n_near_sensitive_site&over_threshold&is_essential]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2.7777777777777777, 0.24935220877729622, 2, array([[12., 12., 16.],\n",
       "        [18., 18., 24.]]))"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "obs = np.array([[10, 10, 20], [20, 20, 20]])\n",
    "stats.chi2_contingency(obs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'annot_pos_list': [258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 291, 440, 459, 488, 489, 496, 497, 498], 'annots': {'metal': [], 'Nucl': [259, 260, 261, 265, 266, 267, 268, 269, 270, 271, 488, 489, 496, 497, 498], 'Site': [], 'DNA_bind': [], 'Ca_bind': [], 'bind_site': [291, 440, 459], 'act_site': [], 'motif': [258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268]}}\n"
     ]
    }
   ],
   "source": [
    "print(dict_of_annot['YOR168W'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number</th>\n",
       "      <th>orf_type</th>\n",
       "      <th>target</th>\n",
       "      <th>position</th>\n",
       "      <th>strand</th>\n",
       "      <th>sequence</th>\n",
       "      <th>mut1_C&gt;G</th>\n",
       "      <th>mut1_C&gt;T</th>\n",
       "      <th>mut2_C&gt;G</th>\n",
       "      <th>mut2_C&gt;T</th>\n",
       "      <th>...</th>\n",
       "      <th>mut_2_effect</th>\n",
       "      <th>mut_2_proximity</th>\n",
       "      <th>co_edit_risk</th>\n",
       "      <th>z_score_meanSE</th>\n",
       "      <th>over_threshold</th>\n",
       "      <th>mut1_perturb_annot</th>\n",
       "      <th>mut2_perturb_annot</th>\n",
       "      <th>mut1_near_annot</th>\n",
       "      <th>mut2_near_annot</th>\n",
       "      <th>perturb_annot_info</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>guide_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>chrXV_650081r</td>\n",
       "      <td>33725</td>\n",
       "      <td>essential</td>\n",
       "      <td>YOR168W</td>\n",
       "      <td>781</td>\n",
       "      <td>-</td>\n",
       "      <td>ACCAATATGCAAATATCCAT</td>\n",
       "      <td>G|267|R</td>\n",
       "      <td>G|267|S</td>\n",
       "      <td>G|267|A</td>\n",
       "      <td>G|267|D</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-4.228881</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>{'metal': [], 'Nucl': [259, 260, 261, 265, 266...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chrXV_650172</td>\n",
       "      <td>33728</td>\n",
       "      <td>essential</td>\n",
       "      <td>YOR168W</td>\n",
       "      <td>869</td>\n",
       "      <td>+</td>\n",
       "      <td>TGACGATACCAACCCCGAAA</td>\n",
       "      <td>D|291|E</td>\n",
       "      <td>silent|GAC&gt;GAT|D|291</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.804775</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>{'metal': [], 'Nucl': [259, 260, 261, 265, 266...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chrXV_650488r</td>\n",
       "      <td>33735</td>\n",
       "      <td>essential</td>\n",
       "      <td>YOR168W</td>\n",
       "      <td>1188</td>\n",
       "      <td>-</td>\n",
       "      <td>ATTCTTAAGATAGCTTCACC</td>\n",
       "      <td>R|402|T</td>\n",
       "      <td>R|402|K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.685215</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>{'metal': [], 'Nucl': [259, 260, 261, 265, 266...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chrXV_650777r</td>\n",
       "      <td>33746</td>\n",
       "      <td>essential</td>\n",
       "      <td>YOR168W</td>\n",
       "      <td>1477</td>\n",
       "      <td>-</td>\n",
       "      <td>TCTTTTGGATAAAACTGTAC</td>\n",
       "      <td>R|499|T</td>\n",
       "      <td>R|499|K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.028205</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>{'metal': [], 'Nucl': [259, 260, 261, 265, 266...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chrXV_650942</td>\n",
       "      <td>33749</td>\n",
       "      <td>essential</td>\n",
       "      <td>YOR168W</td>\n",
       "      <td>1639</td>\n",
       "      <td>+</td>\n",
       "      <td>CCACAAACATTCAAGTTGTG</td>\n",
       "      <td>T|548|R</td>\n",
       "      <td>T|548|I</td>\n",
       "      <td>silent|ACC&gt;ACG|T|547</td>\n",
       "      <td>silent|ACC&gt;ACT|T|547</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-2.477618</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>{'metal': [], 'Nucl': [259, 260, 261, 265, 266...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chrXV_650988r</td>\n",
       "      <td>33751</td>\n",
       "      <td>essential</td>\n",
       "      <td>YOR168W</td>\n",
       "      <td>1688</td>\n",
       "      <td>-</td>\n",
       "      <td>ATCTCGGTGTGGTATCTTCT</td>\n",
       "      <td>R|569|T</td>\n",
       "      <td>R|569|K</td>\n",
       "      <td>silent|CCG&gt;CCC|P|568</td>\n",
       "      <td>silent|CCG&gt;CCA|P|568</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.792031</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>{'metal': [], 'Nucl': [259, 260, 261, 265, 266...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               number   orf_type   target  position strand  \\\n",
       "guide_name                                                   \n",
       "chrXV_650081r   33725  essential  YOR168W       781      -   \n",
       "chrXV_650172    33728  essential  YOR168W       869      +   \n",
       "chrXV_650488r   33735  essential  YOR168W      1188      -   \n",
       "chrXV_650777r   33746  essential  YOR168W      1477      -   \n",
       "chrXV_650942    33749  essential  YOR168W      1639      +   \n",
       "chrXV_650988r   33751  essential  YOR168W      1688      -   \n",
       "\n",
       "                           sequence mut1_C>G              mut1_C>T  \\\n",
       "guide_name                                                           \n",
       "chrXV_650081r  ACCAATATGCAAATATCCAT  G|267|R               G|267|S   \n",
       "chrXV_650172   TGACGATACCAACCCCGAAA  D|291|E  silent|GAC>GAT|D|291   \n",
       "chrXV_650488r  ATTCTTAAGATAGCTTCACC  R|402|T               R|402|K   \n",
       "chrXV_650777r  TCTTTTGGATAAAACTGTAC  R|499|T               R|499|K   \n",
       "chrXV_650942   CCACAAACATTCAAGTTGTG  T|548|R               T|548|I   \n",
       "chrXV_650988r  ATCTCGGTGTGGTATCTTCT  R|569|T               R|569|K   \n",
       "\n",
       "                           mut2_C>G              mut2_C>T  ... mut_2_effect  \\\n",
       "guide_name                                                 ...                \n",
       "chrXV_650081r               G|267|A               G|267|D  ...          NaN   \n",
       "chrXV_650172                    NaN                   NaN  ...          NaN   \n",
       "chrXV_650488r                   NaN                   NaN  ...          NaN   \n",
       "chrXV_650777r                   NaN                   NaN  ...          NaN   \n",
       "chrXV_650942   silent|ACC>ACG|T|547  silent|ACC>ACT|T|547  ...          NaN   \n",
       "chrXV_650988r  silent|CCG>CCC|P|568  silent|CCG>CCA|P|568  ...          NaN   \n",
       "\n",
       "              mut_2_proximity  co_edit_risk z_score_meanSE  over_threshold  \\\n",
       "guide_name                                                                   \n",
       "chrXV_650081r               0           4.0      -4.228881             Yes   \n",
       "chrXV_650172                0           1.0      -1.804775             Yes   \n",
       "chrXV_650488r               0           1.0      -1.685215             Yes   \n",
       "chrXV_650777r               0           1.0      -2.028205             Yes   \n",
       "chrXV_650942                0           3.0      -2.477618             Yes   \n",
       "chrXV_650988r               0           2.0      -1.792031             Yes   \n",
       "\n",
       "               mut1_perturb_annot  mut2_perturb_annot  mut1_near_annot  \\\n",
       "guide_name                                                               \n",
       "chrXV_650081r                 1.0                 1.0              1.0   \n",
       "chrXV_650172                  1.0                 0.0              1.0   \n",
       "chrXV_650488r                 0.0                 0.0              0.0   \n",
       "chrXV_650777r                 0.0                 0.0              1.0   \n",
       "chrXV_650942                  0.0                 0.0              0.0   \n",
       "chrXV_650988r                 0.0                 0.0              0.0   \n",
       "\n",
       "               mut2_near_annot  \\\n",
       "guide_name                       \n",
       "chrXV_650081r              1.0   \n",
       "chrXV_650172               0.0   \n",
       "chrXV_650488r              0.0   \n",
       "chrXV_650777r              0.0   \n",
       "chrXV_650942               0.0   \n",
       "chrXV_650988r              0.0   \n",
       "\n",
       "                                              perturb_annot_info  \n",
       "guide_name                                                        \n",
       "chrXV_650081r  {'metal': [], 'Nucl': [259, 260, 261, 265, 266...  \n",
       "chrXV_650172   {'metal': [], 'Nucl': [259, 260, 261, 265, 266...  \n",
       "chrXV_650488r  {'metal': [], 'Nucl': [259, 260, 261, 265, 266...  \n",
       "chrXV_650777r  {'metal': [], 'Nucl': [259, 260, 261, 265, 266...  \n",
       "chrXV_650942   {'metal': [], 'Nucl': [259, 260, 261, 265, 266...  \n",
       "chrXV_650988r  {'metal': [], 'Nucl': [259, 260, 261, 265, 266...  \n",
       "\n",
       "[6 rows x 41 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp_data[(exp_data['target']=='YOR168W')&(exp_data['over_threshold'] == 'Yes')].head(10)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
